packages/extension-ui/src/Popup/ImportLedger.tsx

```diff
@@ -0,0 +1,177 @@
+    value
+  })));
+
+  const networkOps = useRef(
+    [{
+      text: 'Select network',
+      value: ''
+    },
+    ...ledgerChains.map(({ displayName, genesisHash }): NetworkOption => ({
+      text: displayName,
+      value: genesisHash[0]
+    }))]
+  );
+
+  const _onSave = useCallback(
+    () => {
+      if (address && genesis) {
+        setIsBusy(true);
+
+        createAccountHardware(address, 'ledger', accountIndex, addressOffset, name, genesis)
+          .then(() => onAction('/'))
+          .catch((error: Error) => {
+            console.error(error);
+
+            setIsBusy(false);
+            setError(error.message);
+          });
+      }
+    },
+    [accountIndex, address, addressOffset, genesis, name, onAction]
+  );
+
+  // select element is returning a string
+  const _onSetAccountIndex = useCallback((value: number) => setAccountIndex(Number(value)), []);
+  const _onSetAddressOffset = useCallback((value: number) => setAddressOffset(Number(value)), []);
+
+  return (
+    <>
+      <Header
+        showBackArrow
+        text={t<string>('Import Ledger Account')}
+      />
+      <div className={className}>
+        <Address
+          address={address}
+          genesisHash={genesis}
+          isExternal
+          isHardware
+          name={address ? name : ''}
+        />
+        <Dropdown
+          className='network'
+          label={t<string>('Network')}
+          onChange={setGenesis}
+          options={networkOps.current}
+          value={genesis}
+        />
+        {genesis && address && !ledgerError && (
+          <>
+            <Dropdown
+              className='accountType'
+              label={t<string>('account type')}
+              onChange={_onSetAccountIndex}
+              options={accOps.current}
+              value={accountIndex}
+            />
+            <Dropdown
+              className='accountIndex'
+              label={t<string>('address index')}
+              onChange={_onSetAddressOffset}
+              options={addOps.current}
+              value={addressOffset}
+            />
+          </>
+        )}
+        {!!ledgerWarning && (
+          <Warning>
+            {ledgerWarning}
+          </Warning>
+        )}
+        {(!!error || !!ledgerError) && (
+          <Warning
+            isDanger
+          >
+            {error || ledgerError}
+          </Warning>
+        )}
+      </div>
+      <VerticalSpace/>
+      <ButtonArea>
+        {ledgerLocked
+          ? (<Button
+            isBusy={ledgerLoading || isBusy}
+            onClick={refresh}
+          >
+            <FontAwesomeIcon
+              icon={faSync}
+            />
+            {t<string>('Refresh')}
+          </Button>
+          )
+          : (<Button
+            isBusy={ledgerLoading || isBusy}
+            isDisabled={!!error || !!ledgerError || !address || !genesis}
+            onClick={_onSave}
+          >
+            {t<string>('Import Account')}
+          </Button>)
```

Comment:
This form (which you mostly use) feels weird to me, but maybe just because of what I'm used to. Since in this form you don't need the `()`'s actually. The work better in this form

```js
(
  <Button
    ...
  />
)
```

Reply:
I'm constantly on the fence with these, although I thought I was clear. it's like with 
```js
useCallback(() => {
},[ ]);
```

vs

```js
useCallback(
  () => {
  },
  [ ]
);
```